var/spack/repos/builtin/packages/babelstream/package.py

```diff
@@ -273,123 +481,125 @@ def cmake_args(self):
                 args.append("-DSYCL_COMPILER=ONEAPI-ICPX")
             else:
                 args.append(
-                    "-DSYCL_COMPILER=" + self.spec.variants["implementation"].value.upper()
+                    "-DSYCL_COMPILER=" + self.spec.variants["sycl_compiler_implementation"].value.upper()
                 )
-                if self.spec.variants["implementation"].value.upper() != "ONEAPI-DPCPP":
+                if self.spec.variants["sycl_compiler_implementation"].value.upper() != "ONEAPI-DPCPP":
                     args.append(
-                        "-DSYCL_COMPILER_DIR=" + self.spec.variants["implementation"].value.upper()
+                        "-DSYCL_COMPILER_DIR=" + self.spec.variants["sycl_compiler_implementation"].value.upper()
                     )
-                    if self.spec.variants["implementation"].value.upper() == "COMPUTE-CPP":
+                    if self.spec.variants["sycl_compiler_implementation"].value.upper() == "COMPUTE-CPP":
                         args.append("-DOpenCL_LIBRARY=")
 
         # ===================================
         #             HIP(ROCM)
         # ===================================
 
-        if "+rocm" in self.spec:
-            hip_comp = self.spec["rocm"].prefix + "/bin/hipcc"
+        if "+hip" in self.spec:
+            hip_comp = self.spec["hip"].prefix + "/bin/hipcc"
             args.append("-DCMAKE_CXX_COMPILER=" + hip_comp)
-            args.append(
-                "-DCXX_EXTRA_FLAGS= --offload-arch="
-                + self.spec.variants["amdgpu_target"].value
-                + " "
-                + self.spec.variants["flags"].value
-                + " -O3"
-            )
+            args.append(f'-DCXX_EXTRA_FLAGS= --offload-arch={str(self.spec.variants["amdgpu_target"].value[0])} -O3')
+            if str(self.spec.variants["hip_mem_mode"].value)!="none":
+                args.append("-DMEM=" + self.spec.variants["hip_mem_mode"].value.upper())
 
         # ===================================
         #             TBB
         # ===================================
 
         if "+tbb" in self.spec:
-            args.append("-DONE_TBB_DIR=" + self.spec["tbb"].prefix + "/tbb/latest/")
-            args.append("-DPARTITIONER=" + self.spec.variants["partitioner"].value.upper())
+            args.append("-DONE_TBB_DIR=" + self.spec["intel-tbb"].prefix + "/tbb/latest/")
+            args.append("-DCXX_EXTRA_FLAGS=-ltbb")
+            args.append("-DPARTITIONER=" + self.spec.variants["tbb_partitioner"].value.upper())
+            if self.spec.satisfies("+tbb_use_vector"):
+                args.append("-DUSE_VECTOR=ON")
 
         # ===================================
         #             OpenCL (ocl)
         # ===================================
         if "+ocl" in self.spec:
-            if "backend" in self.spec.variants:
-                if "cuda" in self.spec.variants["backend"].value:
+            if "ocl_backend" in self.spec.variants:
```

Comment:
Doesn't the structure of the variants guarantee this is True if the line above is?

Reply:
Yes, thank you for spotting, I found a few more cases like this I got rid of them too